main/src/ca/uwaterloo/flix/api/lsp/LanguageServer.scala

```diff
@@ -338,15 +338,40 @@ class LanguageServer(port: Int, o: Options) extends WebSocketServer(new InetSock
             ("id" -> requestId) ~ ("status" -> "failure") ~ ("time" -> e) ~ ("result" -> results.map(_.toJSON))
           }
 
-        case failure =>
-          // Case 2: Compilation failed. Send back the error messages.
+        case SoftFailure(root, errors) =>
```

Comment:
Gasp! Code duplication.. Good Henrik is not around.

What we use an or pattern `Success(...) | SoftFailure(...)` would that work?

Reply:
Haha yes! 🙈 

This was just a sketch to see if it was even the correct thing to do. I'll refactor it next year! 😄

Reply:
But in all seriousness I'll find a prettier solution :)

Reply:
I think and OR pattern could work.

Reply:
I think an OR pattern has ambiguous variables that would be bound (`errors` could refer to both patterns) but I'll come up with some other way :)

Reply:
I thought that was the point. Does Scala not allow it?

Reply:
Nope, that's also why we also just use `failure` as the error pattern